src/services/useDebounce.tsx

```diff
@@ -1,14 +1,18 @@
+import { isEqual } from 'lodash';
 import { useEffect, useState } from 'react';
 
 function useDebounce<T>(value: T, delay?: number): T {
   const [debouncedValue, setDebouncedValue] = useState<T>(value);
 
   useEffect(() => {
-    const timer = setTimeout(() => setDebouncedValue(value), delay || 500);
+    // We need to make sure that we compare-deep here as the default useEffect deps do not.
```

Comment:
Re "compare-deep": Why the hyphen? Do you mean "deps"?

Reply:
OK, I see the concept used in JavaScript. I'll so no more about he hyphen. Thank you.